File: MiniDOMBase/DMD/Anagrafica/CarteDiCredito/CarteDiCredito.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases
Imports minidom.Internals
Imports minidom.Anagrafica

Namespace Internals

    Public Class CCarteDiCreditoClass
        Inherits CModulesClass(Of CartaDiCredito)

        Public Sub New()
            MyBase.New("modCarteDiCredito", GetType(CartaDiCreditoCursor), 0)
        End Sub

        Public Function GetItemByName(ByVal value As String) As CartaDiCredito
            value = Strings.Trim(value)
            If (value = "") Then Return Nothing
            Dim cursor As New CartaDiCreditoCursor
            cursor.Stato.Value = ObjectStatus.OBJECT_VALID
            cursor.Name.Value = value
            Dim ret As CartaDiCredito = cursor.Item
            cursor.Dispose()
            Return ret
        End Function

        Public Function GetItemByNumero(ByVal circuito As String, ByVal numeroCarta As String) As CartaDiCredito
            circuito = Strings.Trim(circuito)
            numeroCarta = Strings.Trim(numeroCarta)
            If (circuito = "" OrElse numeroCarta = "") Then Return Nothing
            Dim cursor As New CartaDiCreditoCursor
            cursor.Stato.Value = ObjectStatus.OBJECT_VALID
            cursor.CircuitoCarta.Value = circuito
            cursor.NumeroCarta.Value = numeroCarta
            cursor.DataInizio.SortOrder = SortEnum.SORT_DESC
            Dim ret As CartaDiCredito = cursor.Item
            cursor.Dispose()
            Return ret
        End Function


    End Class
End Namespace


Partial Public Class Anagrafica

    Private Shared m_CarteDiCredito As CCarteDiCreditoClass = Nothing

    Public Shared ReadOnly Property CarteDiCredito As CCarteDiCreditoClass
        Get
            If m_CarteDiCredito Is Nothing Then m_CarteDiCredito = New CCarteDiCreditoClass
            Return m_CarteDiCredito
        End Get
    End Property
 

End Class

File: MiniDOMBase/DMD/Anagrafica/CarteDiCredito/CarteDiCredito.cs
using System;
using DMD;
using DMD.Databases;
using DMD.Databases.Collections;
using static minidom.Sistema;
using static minidom.Anagrafica;
using minidom.repositories;

namespace minidom
{
    namespace repositories
    {

        /// <summary>
        /// Repository di oggetti di tipo <see cref="CartaDiCredito"/>
        /// </summary>
        [Serializable]
        public class CCarteDiCreditoClass 
            : CModulesClass<Anagrafica.CartaDiCredito>
        {

            /// <summary>
            /// Costruttore
            /// </summary>
            public CCarteDiCreditoClass() 
                : base("modCarteDiCredito", typeof(Anagrafica.CartaDiCreditoCursor), 0)
            {
            }

            /// <summary>
            /// Restituisce l'elemento in base al nome
            /// </summary>
            /// <param name="value"></param>
            /// <returns></returns>
            public Anagrafica.CartaDiCredito GetItemByName(string value)
            {
                value = DMD.Strings.Trim(value);
                if (string.IsNullOrEmpty(value))
                    return null;

                using (var cursor = new Anagrafica.CartaDiCreditoCursor())
                {
                    cursor.Stato.Value = ObjectStatus.OBJECT_VALID;
                    cursor.Name.Value = value;
                    return cursor.Item;
                }
            }

            /// <summary>
            /// Restituisce la carta di credito corrente corrispondente al numero
            /// </summary>
            /// <param name="circuito"></param>
            /// <param name="numeroCarta"></param>
            /// <returns></returns>
            public Anagrafica.CartaDiCredito GetItemByNumero(string circuito, string numeroCarta)
            {
                circuito = DMD.Strings.Trim(circuito);
                numeroCarta = DMD.Strings.Trim(numeroCarta);
                if (string.IsNullOrEmpty(circuito) || string.IsNullOrEmpty(numeroCarta))
                    return null;

                using (var cursor = new Anagrafica.CartaDiCreditoCursor())
                {
                    cursor.Stato.Value = ObjectStatus.OBJECT_VALID;
                    cursor.CircuitoCarta.Value = circuito;
                    cursor.NumeroCarta.Value = numeroCarta;
                    cursor.DataInizio.SortOrder = SortEnum.SORT_DESC;
                    return cursor.Item;
                }
            }
        }
    }

    public partial class Anagrafica
    {
        private static CCarteDiCreditoClass m_CarteDiCredito = null;

        /// <summary>
        /// Repository di oggetti di tipo <see cref="CartaDiCredito"/>
        /// </summary>
        public static CCarteDiCreditoClass CarteDiCredito
        {
            get
            {
                if (m_CarteDiCredito is null)
                    m_CarteDiCredito = new CCarteDiCreditoClass();
                return m_CarteDiCredito;
            }
        }
    }
}